local/client/cli/cli.go

```diff
@@ -0,0 +1,45 @@
+package cli
+
+import (
+	"github.com/scootdev/scoot/local/client/conn"
+	"github.com/spf13/cobra"
+)
+
+type CliClient struct {
+	rootCmd *cobra.Command
+	dialer  conn.Dialer
+	conn    conn.Conn
+}
+
+func (c *CliClient) Exec() error {
+	return c.rootCmd.Execute()
+}
+
+func (c *CliClient) Close() error {
+	return c.conn.Close()
+}
+
+func (c *CliClient) openConn() (conn.Conn, error) {
+	if c.conn == nil {
+		conn, err := c.dialer.Dial()
+		if err != nil {
+			return nil, err
+		}
+		c.conn = conn
+	}
+	return c.conn, nil
+}
+
+func NewCliClient(dialer conn.Dialer) (*CliClient, error) {
```

Comment:
I'd want dialer to be optional when making a client. This separation is useful for testing but I'm not a big fan of having to construct trivial defaults when the lib can just do that for me. Also applies to NewServer().